Src/Fluent.sublime-syntax

```diff
@@ -14,53 +14,74 @@ contexts:
   # Fluent's grammar is quite strict about where comments can be placed, but we allow them
   # almost everywhere, since otherwise we're not able to test our syntax definition.
   prototype:
+  - match: ^(?=#)
+    push: comments
+
+  comments:
   # Order matters.
-  - match: ^(###)([^ \n]?)(.*\n?)
+  - match: (###)([^ \n]?)(.*\n?)
     captures:
       1: comment.line.number-sign.resource.fluent punctuation.definition.comment.fluent
       2: invalid.illegal.space-required.fluent
       3: comment.line.number-sign.resource.fluent
-  - match: ^(##)([^ \n]?)(.*\n?)
+    pop: true
+  - match: (##)([^ \n]?)(.*\n?)
     captures:
       1: comment.line.number-sign.group.fluent punctuation.definition.comment.fluent
       2: invalid.illegal.space-required.fluent
       3: comment.line.number-sign.group.fluent
-  - match: ^(#)([^ \n^]?)(.*\n?) # '^' is allowed since it is used in Sublime syntax tests.
+    pop: true
+  - match: (#)([^ \n^]?)(.*\n?) # '^' is allowed since it is used in Sublime syntax tests.
     captures:
       1: comment.line.number-sign.fluent punctuation.definition.comment.fluent
       2: invalid.illegal.space-required.fluent
       3: comment.line.number-sign.fluent
+    pop: true
 
   main:
-  - include: messages
-  - include: terms
+  - match: '{{new_entry_start}}'
+    push: messages-or-terms
   - match: (.).*
     captures:
       1: invalid.illegal.fluent
 
+  messages-or-terms:
+  - include: messages
+  - include: terms
+  - match: ''
+    pop: true
+
+  messages-or-terms-allow-indentation:
+  - match: ^\s*(?=-?{{identifier}}\s*=)
+    set: messages-or-terms
+    embed: messages-or-terms
+    escape: ^(?=\s*(?:-?{{identifier}}\s*=|#))
```

Comment:
Could you explain how this code works, please? [Docs](https://www.sublimetext.com/docs/3/syntax.html) say `set` and `embed` are mutually exclusive.

Reply:
Ah that was an oversight on my part - I tried with `set` originally but couldn't get it to highlight correctly and later switched to `embed` but apparently forgot to comment the `set` line out - they are indeed mutually exclusive - ST just ignores the `set` when there is an `embed` I think, sorry for the confusion